hikari/messages.py

```diff
@@ -309,6 +309,18 @@ class MessageReference:
     a guild.
     """
 
+    def make_link(self) -> str:
+        """Generate a jump link to the referenced message or channel (in case for follow add messages).
+
+        Returns
+        -------
+        str
+            The jump link to the message reference target.
+        """
+        guild_id_str = "@me" if self.guild_id is None else self.guild_id
+        message_id_str = "" if self.id is None else self.id
+        return f"{urls.BASE_URL}/channels/{guild_id_str}/{self.channel_id}/{message_id_str}"
```

Comment:
The behaviour of falling back to just making a channel link here r33lw a bit weird to say the least.

For one it gives weirdly formatted channel links which in-client will show as having a random trailing `/`.
And for the other point I don't really see the logic in doing that for a "message reference link", feels like it'd make more sense to just return None in that case to indicate the message can't be linked to.

Reply:
The only case where id is None is for **Channel** Follow Add messages, so I think it does make sense to fallback to only the channel link then.
Also the extra / is an oversight, I assumed discord would handle that correctly but alas

Reply:
I would personally still consider having separate `make_message_link` and `make_channel_link` methods just to keep it more explicit (it's only really called `make_link` on message since it's linking to the message itself)

But also these can just be "link" properties in a similar sense to the _url properties since they don't take any arguments.

Reply:
Done